### DISCUSSION TITLE
Confused why substructure not matching?

### DISCUSSION THREAD

**Discussion by martinuncountable** [rdkit/rdkit#6715] [Q&A]
I have caffeine:
<img width="560" alt="Screenshot 2023-09-12 at 8 15 15 AM" src="https://github.com/rdkit/rdkit/assets/89403184/c1c47f4b-d612-45d2-8017-e6793d9406a3">

If I remove the double bonded oxygen and then run a substructure match it doesn't work:
<img width="547" alt="Screenshot 2023-09-12 at 8 16 15 AM" src="https://github.com/rdkit/rdkit/assets/89403184/4b72118e-bacb-422f-a767-475e221f1618">

Why is that? If I do the same substructure search on ChEBI https://www.ebi.ac.uk/chebi it returns caffiene

If I enable these settings it seems to work
<img width="544" alt="Screenshot 2023-09-12 at 8 16 53 AM" src="https://github.com/rdkit/rdkit/assets/89403184/8ae84e22-8d35-4820-9e00-4c0a7423c3f3">

Reference colab: https://colab.research.google.com/drive/1vZrnOZLfMXqNYckM0nfoK8TdzRPydAJc#scrollTo=_KKpwElyV6JH

**Comment by bp-kelley** [comment]
In this case, the reason is simple, in one case you have aromatic atoms and in the other you don't.  When you re dealing with "molecules" and not smarts, this implicit aromaticity perception is always in place.

```
smi = 'C1(N(C)C2=C(C(=O)N1C)N(C)C=N2)=O'
smi2 = 'N1(C2CN(C)C(N(C)C=2N=C1)=O)C'
m1 = Chem.MolFromSmiles(smi)
m2 = Chem.MolFromSmiles(smi2)
aromatic = Chem.MolFromSmarts("[a]")
m1.GetSubstructMatches(aromatic)
m2.GetSubstructMatches(aromatic)
```

```
m1
```
![image](https://github.com/rdkit/rdkit/assets/9888416/14d55012-8e26-4a94-b0e1-8fd39ecaff18)

```
m2
```

![image](https://github.com/rdkit/rdkit/assets/9888416/e7a9b20f-d588-4288-b82d-3e3bbe06a64c)
